src/gui/torrentmodel.cpp

```diff
@@ -258,6 +264,23 @@ bool TorrentModel::setData(const QModelIndex &index, const QVariant &value, int
     case TR_CATEGORY:
         torrent->setCategory(value.toString());
         break;
+    case TR_TAGS: {
```

Comment:
The my main objections are related to `setData()` method:
1. I don't know why `Qt::DisplayRole` expected here (this is very old code) although `Qt::EditRole` should be used for simple editing purposes.
2. I do not see the point of supporting `setData()` at all, because we do not use the standard Qt Model/View editing features. We can refer directly to underlying model from the caller code. It will also spare us from having to convert data to/from QVariant.
I'm not asking you to edit existing code, but you should get rid of your code here.